### PR TITLE
LVUIDF-56: rewrite Devise' views to React components

### PR DESCRIPTION
**Pull request by yurko**
https://ssu-jira.softserveinc.com/browse/LVUIDF-56

- [x] Rewrite Devise' views to React components

To review:
- [ ] @rtriska
- [ ] @forecome 

Optional:
- [ ] @irynar

To merge:
- [ ] @izagor
- [ ] @forecome

### CODE REVIEW

File: app/assets/javascripts/components/devise/sign_out/SignOutLink.js.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    e.preventDefault();
+    $.ajax({
+      url: "/users/sign_out",
+      method: "DELETE",
+      success: function() {
+        // ugly redirect, but can't use ReactRouter.Navigation mixin here
+        window.location.href = "/"
+      }.bind(this),
+      error: function(xhr, status, error) {
+        console.error(error);
```
Comment (yurakuzh) [inline app/assets/javascripts/components/devise/sign_out/SignOutLink.js.jsx]: wtf?

File: app/assets/javascripts/components/devise/sign_out/SignOutLink.js.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+var SignOutLink = React.createClass({
+
+  onClick: function(e) {
```
Comment (yurakuzh) [inline app/assets/javascripts/components/devise/sign_out/SignOutLink.js.jsx]: onClick name is confusing, actually you are doing ajax call

Comment (yurko) [inline app/assets/javascripts/components/devise/sign_out/SignOutLink.js.jsx]: it's in progress. will be replaced to a button instead of a link

**Comment by rtriska** [comment]
:+1:

**Pull request merged**